lib/livebook/intellisense.ex

```diff
@@ -519,14 +522,33 @@ defmodule Livebook.Intellisense do
     """
   end
 
+  defp format_definition_link(module, function_or_type \\ nil) do
+    path =
+      :code.get_path()
+      |> Enum.find(&(to_string(&1) =~ "livebook_runtime"))
+      |> to_string()
+      |> Path.join("#{module}.beam")
+
+    if File.exists?(path) do
+      query =
+        case function_or_type do
+          {:function, fun, arity} ->
+            %{module: module_name(module), function: fun, arity: arity}
+
+          {:type, type, arity} ->
+            %{module: module_name(module), type: type, arity: arity}
+
+          nil ->
+            %{module: module_name(module)}
+        end
+
+      "[Go to definition](./go-to-definition?#{URI.encode_query(query)})"
```

Comment:
Is a new URL the best way to go about it? Wouldn't this cause the browser to navigate away from the page when clicked?

Reply:
I tried it locally and I guess this gets hijacked somehow. :) I wonder though that, instead of a URL, we could do a pushEvent over the LiveView, instead of defining a URL?